formatter/formatter.go

```diff
@@ -166,25 +144,27 @@ func (r Reply) WithError(err error) Reply {
 
 // Render renders the message returning the rendered text, or an error if something goes wrong.
 func (r Reply) Render() (string, error) {
-	switch r.mode {
-	case template.Handshake:
-		return r.templates.Build().RenderHandshake(r.to.UserLink)
-	case template.UnknownCommand:
-		return r.templates.Build().RenderUnknownCommand(r.to.UserLink, r.output)
-	case template.Unauthorized:
-		return r.templates.Build().RenderUnauthorizedCommand(r.to.UserLink, r.output, r.err)
-	case template.Failure:
-		return r.templates.Build().RenderFailure(r.to.UserLink, r.err.Error(), r.output)
-	case template.Success:
-		return r.templates.Build().RenderSuccess(r.to.UserLink, r.output)
-	default:
-		return "", fmt.Errorf("don't know how to render mode '%s'", r.mode)
-	}
+	payload := make(map[string]interface{})
+	payload["command"] = r.request.Command
+	payload["args"] = strings.Join(r.request.Args, " ")
+
+	payload["user"] = r.request.Username
+	payload["userlink"] = r.request.UserLink
+	payload["userid"] = r.request.UserID
+	payload["channel"] = r.request.Channel
+	payload["channellink"] = r.request.ChannelLink
+	payload["channelid"] = r.request.ChannelID
+	payload["isim"] = r.request.IsIM
+
+	payload["error"] = r.err
+	payload["output"] = r.output
+
+	return r.templates.Build().Render(r.action, payload)
```

Comment:
At some point, I will simply change the messages and instead of cracking the request open we will just send it in and use it. But I don't want to tackle that just yet.